tt_metal/impl/dispatch/kernels/vc_eth_tunneler.cpp

```diff
@@ -0,0 +1,298 @@
+        (get_compile_time_arg_val(41) & 0xFF),
+        (get_compile_time_arg_val(42) & 0xFF),
+        (get_compile_time_arg_val(43) & 0xFF)
+    };
+
+constexpr uint32_t remote_sender_y[MAX_TUNNEL_LANES] =
+    {
+        (get_compile_time_arg_val(34) >> 8) & 0xFF,
+        (get_compile_time_arg_val(35) >> 8) & 0xFF,
+        (get_compile_time_arg_val(36) >> 8) & 0xFF,
+        (get_compile_time_arg_val(37) >> 8) & 0xFF,
+        (get_compile_time_arg_val(38) >> 8) & 0xFF,
+        (get_compile_time_arg_val(39) >> 8) & 0xFF,
+        (get_compile_time_arg_val(40) >> 8) & 0xFF,
+        (get_compile_time_arg_val(41) >> 8) & 0xFF,
+        (get_compile_time_arg_val(42) >> 8) & 0xFF,
+        (get_compile_time_arg_val(43) >> 8) & 0xFF
+    };
+
+constexpr uint32_t remote_sender_queue_id[MAX_TUNNEL_LANES] =
+    {
+        (get_compile_time_arg_val(34) >> 16) & 0xFF,
+        (get_compile_time_arg_val(35) >> 16) & 0xFF,
+        (get_compile_time_arg_val(36) >> 16) & 0xFF,
+        (get_compile_time_arg_val(37) >> 16) & 0xFF,
+        (get_compile_time_arg_val(38) >> 16) & 0xFF,
+        (get_compile_time_arg_val(39) >> 16) & 0xFF,
+        (get_compile_time_arg_val(40) >> 16) & 0xFF,
+        (get_compile_time_arg_val(41) >> 16) & 0xFF,
+        (get_compile_time_arg_val(42) >> 16) & 0xFF,
+        (get_compile_time_arg_val(43) >> 16) & 0xFF
+    };
+
+constexpr DispatchRemoteNetworkType remote_sender_network_type[MAX_TUNNEL_LANES] =
+    {
+        static_cast<DispatchRemoteNetworkType>((get_compile_time_arg_val(34) >> 24) & 0xFF),
+        static_cast<DispatchRemoteNetworkType>((get_compile_time_arg_val(35) >> 24) & 0xFF),
+        static_cast<DispatchRemoteNetworkType>((get_compile_time_arg_val(36) >> 24) & 0xFF),
+        static_cast<DispatchRemoteNetworkType>((get_compile_time_arg_val(37) >> 24) & 0xFF),
+        static_cast<DispatchRemoteNetworkType>((get_compile_time_arg_val(38) >> 24) & 0xFF),
+        static_cast<DispatchRemoteNetworkType>((get_compile_time_arg_val(39) >> 24) & 0xFF),
+        static_cast<DispatchRemoteNetworkType>((get_compile_time_arg_val(40) >> 24) & 0xFF),
+        static_cast<DispatchRemoteNetworkType>((get_compile_time_arg_val(41) >> 24) & 0xFF),
+        static_cast<DispatchRemoteNetworkType>((get_compile_time_arg_val(42) >> 24) & 0xFF),
+        static_cast<DispatchRemoteNetworkType>((get_compile_time_arg_val(43) >> 24) & 0xFF)
+    };
+
+
+constexpr uint32_t test_results_buf_addr_arg = get_compile_time_arg_val(44);
+constexpr uint32_t test_results_buf_size_bytes = get_compile_time_arg_val(45);
+
+// careful, may be null
+tt_l1_ptr uint32_t* const test_results = reinterpret_cast<tt_l1_ptr uint32_t*>(test_results_buf_addr_arg);
+
+constexpr uint32_t timeout_cycles = get_compile_time_arg_val(46);
+constexpr uint32_t inner_stop_mux_d_bypass = get_compile_time_arg_val(47);
+
+void kernel_main() {
+    rtos_context_switch_ptr = (void (*)())RtosTable[0];
+    //noc_init();
```

Comment:
remove this so we don't reintroduce the bug with calling this in the mux/demux path

Reply:
Fixed.

Reply:
Will add tests to run_cpp_fd2_tests.sh in a separate PR.
tunneling tests need to be skipped on N150 boards.